lib/matplotlib/bezier.py

```diff
@@ -316,7 +317,7 @@ def _f(xy):
 
 def get_cos_sin(x0, y0, x1, y1):
     dx, dy = x1 - x0, y1 - y0
-    d = (dx * dx + dy * dy) ** .5
+    d = (dx * dx + dy * dy) ** .5 or 1  # Account for divide by zero
```

Comment:
btw, can we change that expression to use np.hypot()?

On Fri, Feb 27, 2015 at 2:08 PM, Benjamin Root <ben.root@ou.edu> wrote:

> Personally, I like that compact statement when I am selecting one of two
> items: "somevar = foo or bar". Particularly useful for situations like the
> following: "foo, bar = zip(somelist or [[], []])" to prevent errors if
> somelist is empty.
>
> But I agree here... having an "or" towards the end of a mathematical
> expression just seems wrong. I am not looking for an "or" there, and it is
> easy to miss.
>
> On Fri, Feb 27, 2015 at 1:46 PM, Eric Firing <notifications@github.com>
> wrote:
>
>> In lib/matplotlib/bezier.py
>> <https://github.com/matplotlib/matplotlib/pull/4173#discussion_r25528281>
>> :
>>
>> > @@ -316,7 +317,7 @@ def _f(xy):
>> >
>> >  def get_cos_sin(x0, y0, x1, y1):
>> >      dx, dy = x1 - x0, y1 - y0
>> > -    d = (dx * dx + dy * dy) ** .5
>> > +    d = (dx * dx + dy * dy) ** .5 or 1  # Account for divide by zero
>>
>> As a side note, my negative reaction to it reflected my earlier days
>> working with Perl--it's a typical Perl-style construct, compact but a bit
>> tricky, relying on evaluating an expression in two distinct ways (numerical
>> and Boolean) in a single statement.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/matplotlib/matplotlib/pull/4173/files#r25528281>.
>>
>
>